### PR TITLE
Deprecate some functions in Validation

### PR DESCRIPTION
**Pull request by wallymathieu**

### CODE REVIEW

File: src/FSharpPlus/Validations.fs [hunk 1]
```diff
@@ -113,6 +113,7 @@ module Validation=
     ///
     /// validate : 'e -> ('a -> bool) -> 'a -> Validation<'e, 'a>
     ///
+    [<System.Obsolete("The validate method is less generic than intended.")>]
```
Comment (gusty) [inline src/FSharpPlus/Validations.fs]: I don't think we should give that reason for the deprecation. Let's put something like "This function will not be supported in future versions."

**Comment by gusty** [comment]
We can also take the opportunity to rename (and deprecate) both `orElse`  and `valueOr` as they should be aligned with the recent naming (which at the same time is aligned with the existing F# core functions) `defaultValue` and `defaultWith`.

File: src/FSharpPlus/Validations.fs (hunk 1 above)
Comment (wallymathieu) [reply src/FSharpPlus/Validations.fs]: Let's go with that then

File: src/FSharpPlus/Validations.fs [hunk 2]
```diff
@@ -77,12 +77,14 @@ module Validation=
...
 
     /// orElse v a returns 'a when v is Failure, and the a in Success a.
+    [<System.Obsolete("This function will not be supported in future versions.")>]
     let orElse v (a: 'a) = 
         match v with
         | Failure _ -> a
         | Success x -> x
 
     /// Return the 'a or run the given function over the 'e.
+    [<System.Obsolete("This function will not be supported in future versions.")>]
```
Comment (gusty) [inline src/FSharpPlus/Validations.fs]: Great, but for these functions the reason is not the same. We can put here "Use Validation.defaultWith instead."

Comment (wallymathieu) [reply src/FSharpPlus/Validations.fs]: We should add the `defaultWith` and `defaultValue` functions as well

**Comment by gusty** [comment]
All green. Please Squash and Merge at your convenience.

**Comment by wallymathieu** [comment]
Thanks @gusty for the help!

**Pull request merged**